infrasec/tutorials/atlantis.md

```diff
@@ -0,0 +1,418 @@
+
+## Step 3: Call the [Atlantis module](https://registry.terraform.io/modules/terraform-aws-modules/atlantis/aws/latest) 🧜
+
+We'll start with the most basic Atlantis module call we can. Assuming an existing VPC, Robot user, validated certificate, zone name, Atlantis Docker image, and GitHub connection, the code will look something like this
+
+  ```hcl
+  module "atlantis" {
+    source  = "terraform-aws-modules/atlantis/aws"
+    version = "~> 2.36.0" // make sure to update to the latest version
+
+    name = "${var.environment}-atlantis"
+
+    atlantis_image = "milmove/circleci-docker:milmove-atlantis"
+
+    # Route53
+    create_route53_record = false
+    route53_zone_name = "atlantis.exp.move.mil"
+    certificate_arn = "arn:aws-us-gov:acm:us-gov-west-1:015681133840:certificate/456c11bd-a730-476d-ba7c-1bd4690ede3f"
+
+    # VPC
+    vpc_id          = ""
+    cidr            = "10.30.0.0/16"
+    azs             = ["us-gov-east-1a", "us-gov-east-1b", "us-gov-east-1c"]
+    private_subnet_ids = ["subnet-012345678f01", "subnet-012345678f02", "subnet-012345678f03"]
+    public_subnet_ids  = ["subnet-012345678f04", "subnet-012345678f05", "subnet-012345678f06"]
+
+    # Atlantis
+    atlantis_github_user = "atlantis"
+    atlantis_github_user_token = data.aws_ssm_parameter.github_user_token.value
+    atlantis_repo_whitelist          = ["github.com/transcom/transcom-infrasec-gov"]
+    atlantis_hide_prev_plan_comments = "true"
+    allow_github_webhooks = true
+  }
+  ```
+
+Eventually we'll build out until our call looks lot like [the example in `legendary-waddle`](https://github.com/trussworks/legendary-waddle/blob/ce52ce64ac2c41be1c0c8e9b3ed577b968714bd6/trussworks-prod/atlantis-prod/main.tf#L33-L123). Note that the example above is GovCloud-based. Replace any references to `us-gov` with `us` if not using GovCloud.
+
+1. We'll also need to add a little code to pull in the token from AWS as data:
+
+    ```hcl
+    data "aws_ssm_parameter" "github_user_token" {
+      name = "/atlantis/github/user/token"
+    }
+    ```
+
+1. Add custom environment secrets:
+
+    ```hcl
+    # from https://trussworks.slack.com/archives/CLNC1MUBS/p1615847664002200?thread_ts=1615570522.023200&cid=CLNC1MUBS
+      custom_environment_secrets = [
+        {
+          name      = "ATLANTIS_GH_TOKEN"
+          valueFrom = data.aws_ssm_parameter.github_user_token.name
+        },
+        # {
+        #   name      = "ATLANTIS_GH_WEBHOOK_SECRET"
+        #   valueFrom = local.webhook_ssm_parameter_name
+        # },
+      ]
+    ```
+
+1. Submit a PR, get approval, and `terraform apply` the code.
+
+## Step 4: Configure Your Webhook
+
+The Atlantis module conveniently creates a GitHub webhook for you, and the documentation on this is fairly good. Follow the Atlantis documentation on [how to configure that webhook](https://www.runatlantis.io/docs/configuring-webhooks.html#github-github-enterprise).
```

Comment:
What do you want me to get out of this Atlantis docs? Can you summarize what you want out of it in a couple sentences and then say "for more info, see the docs here"?